internal/handler/msg_getmore.go

```diff
@@ -266,3 +273,81 @@ func (h *Handler) makeNextBatch(c *cursor.Cursor, batchSize int64) (*types.Array
 
 	return nextBatch, nil
 }
+
+// awaitData stops the goroutine, and waits for a new data for the cursor.
+// If there's a new document, or the maxTimeMS have passed it returns the nextBatch.
+func (h *Handler) awaitData(ctx context.Context, c *cursor.Cursor, maxTimeMS, batchSize int64) (resBatch *types.Array, err error) {
+	data := c.Data.(*findCursorData)
+
+	closer := iterator.NewMultiCloser()
+
+	startTime := time.Now()
+	sleepDur := time.Duration(maxTimeMS) * time.Millisecond
+
+	ctx, cancel := context.WithTimeout(ctx, sleepDur)
+	defer cancel()
+
+	defer func() {
+		if err == nil {
+			return
+		}
+
+		if errors.Is(err, context.DeadlineExceeded) {
+			err = nil
+			return
+		}
+
+		err = lazyerrors.Error(err)
+		resBatch = nil
+		return
+	}()
+
+	for {
```

Comment:
Don't we need any `time.Sleep` in this `for` loop? It works but the amount of queries it makes is a bit crazy...

Reply:
We can implement it yeah